### PR TITLE
Add statistics functions: skewSamp, skewPop, kurtSamp, and kurtPop

### PR DESCRIPTION
**Pull request by hczhcz**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):

Add statistics functions to compute skewness and kurtosis.

Detailed description (optional):

### CODE REVIEW

File: dbms/src/AggregateFunctions/AggregateFunctionStatisticsSimple.h [hunk 1]
```diff
@@ -139,8 +136,180 @@ struct VarMomentsDecimal
...
+};
+
+template <typename T, size_t _level>
+struct HighOrderMoments
+{
+    T m0{};
+    T m1{};
+    T m2{};
+    T m3{};
+    T m4{};
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/AggregateFunctionStatisticsSimple.h]: It's wrong that we serialize / deserialize / keep in memory value even if it is unneeded.

File: dbms/src/AggregateFunctions/AggregateFunctionStatisticsSimple.h [hunk 2]
```diff
@@ -139,8 +136,180 @@ struct VarMomentsDecimal
...
+            ) * m1 / m0
+        ) / m0;
+    }
+
+    T NO_SANITIZE_UNDEFINED getMoment4() const
+    {
+        // to avoid accuracy problem
+        if (m0 == 1)
+            return 0;
+        return (m4
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/AggregateFunctionStatisticsSimple.h]: Need a link or description.

**Comment by alexey-milovidov** [comment]
The only remaining is to fix UBSan:
https://clickhouse-test-reports.s3.yandex.net/5200/cf95327cba4d764aa3c12a1140e3db43a13c7421/functional_stateless_tests_(ubsan)/stderr.txt

File: dbms/src/AggregateFunctions/AggregateFunctionStatisticsSimple.h [hunk 3]
```diff
@@ -139,8 +136,190 @@ struct VarMomentsDecimal
...
+
+template <typename T, size_t _level>
+struct HighOrderMoments
+{
+    T m[_level + 1]{};
+
+    void add(T x)
+    {
+        ++m[0];
+        m[1] += x;
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/AggregateFunctionStatisticsSimple.h]: And possibly remove struct VarMoments, because this struct can be generalization?

**Comment by alexey-milovidov** [comment]
Ok.

Formulas are not completely generalized; there is no static_assert/enable_if in functions for higher order moments. But still good to merge.

**Comment by BayoNet** [comment]
@hczhcz 

Hello :-).

I'm trying to add documentation for these functions, and I need some help. I have made a rough draft in my fork for these functions description.

Would you please to help me to fill it with valuable information? I am sure you can tell the users much more than I did. For example, I don't understand the difference between Pop and Samp functions.

- https://github/BayoNet/ClickHouse/blob/DOCAPI-6887-skewSamp/docs/en/query_language/agg_functions/reference.md#skewpop
- https://github.com/BayoNet/ClickHouse/blob/DOCAPI-6887-skewSamp/docs/en/query_language/agg_functions/reference.md#skewsamp
- https://github.com/BayoNet/ClickHouse/blob/DOCAPI-6887-skewSamp/docs/en/query_language/agg_functions/reference.md#kurtpop
- https://github.com/BayoNet/ClickHouse/blob/DOCAPI-6887-skewSamp/docs/en/query_language/agg_functions/reference.md#kurtsamp

**Pull request merged**